### ISSUE TITLE
False report of 1015

### ISSUE THREAD

**Issue by wmjordan** [u2uconsult/codeanalyzers#34]
The following code snippet demonstrates the issue:

````C#
var a = new int[0];
for (var i = 0; i < a.Length; i++) {
   Console.WriteLine(true ? a[i] : a[i]);
}
````

If we replace the `?:` with `if...else`, no report, which is correct.

**Comment by KrisVandermotten** [comment]
Thanks for reporting. This should be fixed in release 3.8.3.

**Comment by wmjordan** [comment]
Please reopen this issue, it is not yet fully fixed.
The following code triggers a false report of 1015.
````C#
var a = new int[0];
for (var i = 0; i < a.Length; i++) {
   if (i % 2 == 0) {
    a[i] = 1;
    continue;
   }
   if (i % 3 == 0) {
     a[i] = -1;
     continue;
   }
// do something else not relevant to array `a`
}
````
